Crash when chucker is used with proguard enabled.

Chucker is crashing with these crash logs when clicked on the notification. It is working fine in the debug build when proguard is not enabled but crashing when the proguard is enabled.

2019-11-26 17:39:06.222 18381-18381/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.gojek.driver.bike.staging, PID: 18381
    android.view.InflateException: Binary XML file line #17: Binary XML file line #17: Error inflating class <unknown>
    Caused by: android.view.InflateException: Binary XML file line #17: Error inflating class <unknown>
    Caused by: java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Constructor.newInstance0(Native Method)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
        at android.view.LayoutInflater.createView(LayoutInflater.java:647)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:790)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:730)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:492)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:423)
        at dark.Ɨſ.onCreateView(:65)
        at dark.ιʟ.performCreateView(:2439)
        at dark.Іɹ.ˊ(:1460)
        at dark.Іɹ.ॱॱ(:1784)
        at dark.Іɹ.ॱ(:1852)
        at dark.ϙ.ॱॱ(:802)
        at dark.Іɹ.ˋ(:2625)
        at dark.Іɹ.ˊ(:2411)
        at dark.Іɹ.ˋ(:2366)
        at dark.Іɹ.ˏ(:2243)
        at dark.ϙ.ʼ(:654)
        at dark.іɹ.finishUpdate(:146)
        at dark.ɨӏ.ˊ(:1244)
        at dark.ɨӏ.ˏ(:1092)
        at dark.ɨӏ.onMeasure(:1622)
        at android.view.View.measure(View.java:23355)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6758)
        at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1535)
        at android.widget.LinearLayout.measureVertical(LinearLayout.java:825)
        at android.widget.LinearLayout.onMeasure(LinearLayout.java:704)
        at android.view.View.measure(View.java:23355)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6758)
        at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
        at dark.ɩı.onMeasure(:143)
        at android.view.View.measure(View.java:23355)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6758)
        at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1535)
        at android.widget.LinearLayout.measureVertical(LinearLayout.java:825)
        at android.widget.LinearLayout.onMeasure(LinearLayout.java:704)
        at android.view.View.measure(View.java:23355)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6758)
        at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
        at android.view.View.measure(View.java:23355)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6758)
        at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1535)
        at android.widget.LinearLayout.measureVertical(LinearLayout.java:825)
        at android.widget.LinearLayout.onMeasure(LinearLayout.java:704)
        at android.view.View.measure(View.java:23355)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6758)
        at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
        at com.android.internal.policy.DecorView.onMeasure(DecorView.java:717)
        at android.view.View.measure(View.java:23355)
        at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:2919)
        at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java:1749)
        at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2042)
        at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1637)
        at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7797)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1172)
        at android.view.Choreographer.doCallbacks(Choreographer.java:984)
        at android.view.Choreographer.doFrame(Choreographer.java:809)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1158)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6898)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:537)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
2019-11-26 17:39:06.222 18381-18381/? E/AndroidRuntime: Caused by: java.lang.IllegalStateException: Binary XML file line #17: Unable to find LayoutManager dark.LinearLayoutManager
        at dark.ιɍ.ˊ(:810)
        at dark.ιɍ.<init>(:713)
        at dark.ιɍ.<init>(:647)
        	... 65 more
     Caused by: java.lang.ClassNotFoundException: Didn't find class "dark.LinearLayoutManager" on path: DexPathList[[zip file "/system/framework/org.apache.http.legacy.boot.jar", zip file "/data/app/com.gojek.driver.bike.staging-QEokmeEQhz6U6wW5zueoIg==/base.apk"],nativeLibraryDirectories=[/data/app/com.gojek.driver.bike.staging-QEokmeEQhz6U6wW5zueoIg==/lib/arm64, /data/app/com.gojek.driver.bike.staging-QEokmeEQhz6U6wW5zueoIg==/base.apk!/lib/arm64-v8a, /system/lib64]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:134)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        at dark.ιɍ.ˊ(:791)
        	... 67 more

@vbuberen 

From the crash logs 
Caused by: java.lang.ClassNotFoundException: Didn't find class "dark.LinearLayoutManager" on path

**-repackageclasses "dark"** might be the reason that I was getting this issue.  It repackages all class files and moves them into the single given package.

--- a/library/proguard-rules.pro
+++ b/library/proguard-rules.pro
@@ -1,2 +1,1 @@
 -keep class com.chuckerteam.chucker.internal.data.entity.HttpTransaction { *; }
--keep class android.support.v7.widget.SearchView { *; }

--- a/library/src/main/java/com/chuckerteam/chucker/internal/ui/error/ErrorListFragment.kt
+++ b/library/src/main/java/com/chuckerteam/chucker/internal/ui/error/ErrorListFragment.kt
@@ -15,7 +15,6 @@ import androidx.fragment.app.Fragment
 import androidx.lifecycle.Observer
 import androidx.recyclerview.widget.DividerItemDecoration
 import androidx.recyclerview.widget.DividerItemDecoration.VERTICAL
-import androidx.recyclerview.widget.LinearLayoutManager
 import androidx.recyclerview.widget.RecyclerView
 import com.chuckerteam.chucker.R
 import com.chuckerteam.chucker.internal.data.repository.RepositoryProvider
@@ -37,7 +36,6 @@ internal class ErrorListFragment : Fragment() {
             findViewById<TextView>(R.id.link).movementMethod = LinkMovementMethod.getInstance()
 
             val recyclerView = findViewById<RecyclerView>(R.id.list)
-            recyclerView.layoutManager = LinearLayoutManager(context)
             recyclerView.addItemDecoration(DividerItemDecoration(context, VERTICAL))
             adapter = ErrorAdapter(listener)
             recyclerView.adapter = adapter

--- a/library/src/main/java/com/chuckerteam/chucker/internal/ui/transaction/TransactionListFragment.kt
+++ b/library/src/main/java/com/chuckerteam/chucker/internal/ui/transaction/TransactionListFragment.kt
@@ -32,7 +32,6 @@ import androidx.fragment.app.Fragment
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.Observer
 import androidx.recyclerview.widget.DividerItemDecoration
-import androidx.recyclerview.widget.LinearLayoutManager
 import androidx.recyclerview.widget.RecyclerView
 import com.chuckerteam.chucker.R
 import com.chuckerteam.chucker.internal.data.entity.HttpTransactionTuple
@@ -66,7 +65,6 @@ internal class TransactionListFragment :
 
         val recyclerView = view.findViewById<RecyclerView>(R.id.list)
         val context = view.context
-        recyclerView.layoutManager = LinearLayoutManager(context)
         recyclerView.addItemDecoration(
             DividerItemDecoration(context, DividerItemDecoration.VERTICAL)
         )

--- a/library/src/main/res/layout/chucker_fragment_error_list.xml
+++ b/library/src/main/res/layout/chucker_fragment_error_list.xml
@@ -25,7 +25,7 @@
         android:layout_width="match_parent"
         android:layout_height="match_parent"
         android:scrollbars="vertical"
-        app:layoutManager="LinearLayoutManager"
+        app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
         tools:listitem="@layout/chucker_list_item_error" />
 
     <LinearLayout

--- a/library/src/main/res/layout/chucker_fragment_transaction_list.xml
+++ b/library/src/main/res/layout/chucker_fragment_transaction_list.xml
@@ -25,7 +25,7 @@
         android:layout_width="match_parent"
         android:layout_height="match_parent"
         android:scrollbars="vertical"
-        app:layoutManager="LinearLayoutManager"
+        app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
         tools:context="com.chuckerteam.chucker.internal.ui.transaction.TransactionListFragment"
         tools:listitem="@layout/chucker_list_item_transaction" />
 

